.travis.yml

```diff
@@ -1,19 +1,30 @@
 language: java
-
-script: ./gradlew build
-
 jdk:
   - oraclejdk8
 
+# Anything after trusty doesn't have Java 8
 dist: trusty
 
-#Below skips the installation step completely (https://docs.travis-ci.com/user/customizing-the-build/#Skipping-the-Installation-Step)
-#If we don't skip it Travis runs unnecessary Gradle tasks like './gradlew assemble'
-install:
- - true
+# Below skips the installation step completely (https://docs.travis-ci.com/user/customizing-the-build/#Skipping-the-Installation-Step)
+# If we don't skip it Travis runs unnecessary Gradle tasks like './gradlew assemble'
+install: true
 
-# Only build on master and release tags
+# Build and test each commit on master and each PR to master
+script: ./gradlew build
 branches:
   only:
   - master
-  - /^v\d+\.\d+\.\d+$/
+
+# Publish a new version on tag push
+deploy:
+  provider: script
+  script: scripts/travis/publish-tag.sh
+  on:
+    tags: true
+
+# Send email notifications to the user associated with a tag deployment
```

Comment:
will this send a notification to all contributors? or how are users associated with a tag determined?

Reply:
Nope. According to the [Travis docs](https://docs.travis-ci.com/user/notifications/#how-is-the-build-email-receiver-determined), the recipient is determined using the email associated with the "committer and author", which presumably refers to the commit committer and author. I'm not sure how this will play out for tags; will it choose the commit's author, or the tagger of the tag? Can't say for now. We can also change it to send to an internal mailing list.